spec.md

```diff
@@ -600,6 +612,147 @@ message Volume {
   // the same attributes. This field is OPTIONAL and when present MUST
   // be passed to volume validation and publishing calls.
   map<string,string> attributes = 3;
+
+  // The topological segment(s) the provisioned volume is accessible
+  // from. A topological segment is a sub-division of a cluster, like
+  // region, zone, rack, etc.
+  // Each key must consist of alphanumeric characters, '-', '_' or '.'.
+  // Each value MUST contain 1 or more valid strings.
+  // A value containing more than 1 string indicates the volume is
+  // accessible from multiple topological segments.
+  // COs may use this information along with the topology information
+  // returned by NodeGetId to ensure that a given volume is accessible
+  // from a given node when scheduling workloads.
+  // This field is OPTIONAL. If it is not specified, the CO MAY assume
+  // the volume is equally accessible from all nodes in the cluster and
+  // may schedule workloads referencing the volume on any available node
+  //
+  // Example 1:
+  // Given a volume that is accessible from a single zone, and
+  // accessibility_requirements =
+  //   {"region":{"us-west1"}, "zone":{"us-west1-b"}}
+  // Indicates the volume is accessible only from the "region"
+  // "us-west1" and the "zone" "us-west1-b".
+  //
+  // Example 2:
+  // Given a volume that is accessible from two zones, and
+  // accessibility_requirements =
+  //   {"region": {"us-west1"}, "zone": {"us-west1-b", "us-west1-c"}}
+  // Indicates the volume is accessible from both "zone" "us-west1-b"
+  // and "zone" "us-west1-c" the "region" "us-west1".
+  map<string, ListOfString> accessibility_requirements = 4;
+}
+
+// TopologyRequirement captures the topological segment(s) the CO wants
+// a volume to be provisioned in.
+message TopologyRequirement {
```

Comment:
naming: consider `TopologyPolicy` instead? This message MAY define both requisite and preference topology segment lists, each of which are OPTIONAL. Given that a message of this type may only contain preferences, the suffix "Requirement" seems to overreach.

Reply:
Naming is hard. 

> Given that a message of this type may only contain preferences

I don't think this is accurate. As it is currently worded, the `TopologyRequirement` message, if it exists, would contain the *required* set of possibilities along with which of those are preferred. It does not allow for a list of  preferred options only. As worded anything that is in `preferred_topologies` must be present in `accepted_topologies`. So therefore I think requirement seems more accurate.

However, I could loosen this to allow for preferred only -- so for example `preferred_topologies` could exist without `accepted_topologies` meaning SP should use `preferred_topologies` but if that doesn't work, use whatever it wants. If we do this then `Policy` would be a better name. But I'm not sure we want to enable that use case.